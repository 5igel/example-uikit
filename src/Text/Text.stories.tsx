import React from 'react'
// also exported from '@storybook/react' if you can deal with breaking changes in 6.1
import { Story, Meta } from '@storybook/react/types-6-0';

import { Text, TextProps } from './Text';

export default {
  title: 'Example/Text',
  component: Text,
} as Meta;

const Template: Story<TextProps> = (args) => <Text {...args} />;

export const Heading1 = Template.bind({});
Heading1.args = {
  variant: 'h1',
  children: 'Heading 1 36px light'
};

export const Heading2 = Template.bind({});
Heading2.args = {
  variant: 'h2',
  children: 'Heading 2'
};

export const Heading3 = Template.bind({});
Heading3.args = {
  variant: 'h3',
  children: 'Heading 3'
};

export const Heading4 = Template.bind({});
Heading4.args = {
  variant: 'h4',
  children: 'Heading 4'
};

export const Heading5 = Template.bind({});
Heading5.args = {
  variant: 'h5',
  children: 'Heading 5'
};

export const Heading6 = Template.bind({});
Heading6.args = {
  variant: 'h6',
  children: 'Heading 6'
};

export const Body = Template.bind({});
Body.args = {
  variant: 'body',
  children: 'Body'
};

export const Caption = Template.bind({});
Caption.args = {
  variant: 'caption',
  children: 'Caption'
};
