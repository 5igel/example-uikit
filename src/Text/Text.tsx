import {PropsWithChildren} from 'react';
import styled from '@emotion/styled'
import { variant, typography, color, ColorProps, TypographyProps } from 'styled-system'
import shouldForwardProp from '@styled-system/should-forward-prop'

interface OwnProps {
  variant?: 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6' | 'body' | 'caption'
}

export type TextProps = PropsWithChildren<OwnProps & TypographyProps & ColorProps>;

export const Text = styled('span', {shouldForwardProp})(
  typography,
  color,
  variant({
    variants: {
      h1: {
        fontWeight: 0,
        fontSize: 6,
      },
      h2: {
        fontWeight: 0,
        fontSize: 5,
      },
      h3: {
        fontWeight: 0,
        fontSize: 4,
      },
      h4: {
        fontWeight: 1,
        fontSize: 3,
      },
      h5: {
        fontWeight: 1,
        fontSize: 2,
      },
      h6: {
        fontWeight: 2,
        fontSize: 1,
      },
      body: {
        fontWeight: 1,
        fontSize: 1,
      },
      caption: {
        fontWeight: 1,
        fontSize: 0,
      },
    }
  })
);

Text.defaultProps = {
  variant: 'body'
}
