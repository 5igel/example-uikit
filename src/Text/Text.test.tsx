import React from 'react';
import { render } from '@testing-library/react';

import {Text} from './Text';

test('renders text h1', () => {
  const { container } = render(<Text variant="h1">Heading 1</Text>)
  expect(container.firstChild).toMatchSnapshot()
});

test('renders text h2', () => {
  const { container } = render(<Text variant="h2">Heading 2</Text>)
  expect(container.firstChild).toMatchSnapshot()
});

test('renders text h3', () => {
  const { container } = render(<Text variant="h3">Heading 3</Text>)
  expect(container.firstChild).toMatchSnapshot()
});

test('renders text h4', () => {
  const { container } = render(<Text variant="h4">Heading 4</Text>)
  expect(container.firstChild).toMatchSnapshot()
});

test('renders text h5', () => {
  const { container } = render(<Text variant="h5">Heading 5</Text>)
  expect(container.firstChild).toMatchSnapshot()
});

test('renders text h6', () => {
  const { container } = render(<Text variant="h6">Heading 6</Text>)
  expect(container.firstChild).toMatchSnapshot()
});
