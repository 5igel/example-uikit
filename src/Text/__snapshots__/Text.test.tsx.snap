// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders text h1 1`] = `
<span
  class="css-1puunqr"
>
  Heading 1
</span>
`;

exports[`renders text h2 1`] = `
<span
  class="css-18csn3"
>
  Heading 2
</span>
`;

exports[`renders text h3 1`] = `
<span
  class="css-1r2j1hy"
>
  Heading 3
</span>
`;

exports[`renders text h4 1`] = `
<span
  class="css-tpgt8l"
>
  Heading 4
</span>
`;

exports[`renders text h5 1`] = `
<span
  class="css-i125p7"
>
  Heading 5
</span>
`;

exports[`renders text h6 1`] = `
<span
  class="css-7dkg9j"
>
  Heading 6
</span>
`;
